doc/api/index.rst

```diff
@@ -1,67 +1,49 @@
-API Overview
-============
+API
+===
 
 .. toctree::
    :hidden:
 
    api_changes
 
-.. contents:: :local:
+For recent changes, see :doc:`api_changes`.
 
-See also the :doc:`api_changes`.
+Much of the library is accessed by creating :doc:`Figure <figure_api>` 
+and :doc:`Axes <axes_api>` objects and calling methods on them. 
 
-Usage patterns
---------------
-
-Below we describe several common approaches to plotting with Matplotlib.
-
-The pyplot API
-^^^^^^^^^^^^^^
-
-`matplotlib.pyplot` is a collection of command style functions that make
-Matplotlib work like MATLAB. Each pyplot function makes some change to a
-figure: e.g., creates a figure, creates a plotting area in a figure, plots
-some lines in a plotting area, decorates the plot with labels, etc.
-
-`.pyplot` is mainly intended for interactive plots and simple cases of
-programmatic plot generation.
-
-Further reading:
-
-- The `matplotlib.pyplot` function reference
-- :doc:`/tutorials/introductory/pyplot`
-- :ref:`Pyplot examples <pyplots_examples>`
-
-.. _api-index:
+  - :doc:`figure_api`: axes creation, figure-level artists
+  - :doc:`axes_api`: most plotting methods, axes labels, access to 
+    axis styling, etc.  
 
-The object-oriented API
-^^^^^^^^^^^^^^^^^^^^^^^
+   
+(Note that there is also a parallel `matplotlib.pyplot` API interface 
+that is considered useful for interactive work; see 
+:ref:`usage patterns <usage_patterns>`, below).
```

Comment:
```suggestion
Alternatively, there is also a parallel `matplotlib.pyplot` API interface 
that is considered useful for interactive work; see 
:ref:`usage patterns <usage_patterns>`, below.
```

Reply:
But I'm not sure we should mention the distinction here at all. Particularly also because the following example uses the object-based approach, but still needs to use pyplot for Figure and Axes creation. It's quite confusing that the non-pyplot approach still needs pyplot. So, I'd not go into the details here.

Reply:
Sure - I struggled a bit with how to work this in, or if I should.  It is such a common point of confusion that someone wrote a whole section and stuck it at the top of the API docs, so it seemed at least referring to it near the top would be useful, but I'm happy to leave it out.